ui/src/test/org/jboss/windup/web/selenium/CreateProject.java

```diff
@@ -0,0 +1,1348 @@
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxOptions;
+//import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.htmlunit.HtmlUnitDriver;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.openqa.selenium.firefox.GeckoDriverService;
+import org.jboss.windup.web.selenium.AnalyzeProject.Application;
+import org.openqa.selenium.Alert;
+import com.google.common.base.Preconditions;
+
+import java.awt.AWTException;
+import java.awt.Robot;
+import java.awt.event.KeyEvent;
+import java.io.File;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.logging.Level;
+
+/**
+ * this code is intended for a RHAMT web application that does not have any
+ * pre-made projects
+ * 
+ * @author elise
+ *
+ */
+public class CreateProject {
+
+	private WebDriver driver;
+
+	private WebElement projButton;
+	private WebElement nameInput;
+	private WebElement descInput;
+	private WebElement cancel;
+	private WebElement next;
+	private WebElement chooseFiles;
+	private WebElement fileUpload;
+	private WebElement checkbox;
+
+	public CreateProject() {
+		// Create a new instance of the Firefox driver
+		// Notice that the remainder of the code relies on the interface,
+		// not the implementation.
+		System.setProperty("webdriver.gecko.driver", "/usr/lib/node_modules/geckodriver/bin/geckodriver");
+		System.setProperty("webdriver.chrome.driver","/usr/bin/chromedriver");
+
+		ChromeOptions options = new ChromeOptions();
+		options.setBinary("/usr/bin/chromium-browser"); // Location where Chrome is installed
+		options.addArguments("--headless");
+		options.addArguments("--disable-gpu");
+		options.addArguments("--no-sandbox");
+		options.addArguments("--allow-insecure-localhost");
+		options.addArguments("--networkConnectionEnabled");
+		//options.AddAdditionalCapability();
+		//options.setHeadless(true);
+		driver = new ChromeDriver(options);
+
+		// opens up the browser
+		driver.get("http://127.0.0.1:8080/");
+
+		WebElement header = (new WebDriverWait(driver, 15))
+				.until(ExpectedConditions.presenceOfElementLocated(By.id("header-logo")));
+	}
+
+	/**
+	 * clicks the new project button This is when there are no other projects in the
+	 * project list should re-direct to
+	 * http://127.0.0.1:8080/rhamt-web/wizard/create-project
+	 */
+	public void clickNewProjButton() {
+		WebDriverWait wait = new WebDriverWait(driver, 10);
+		wait.until(ExpectedConditions.elementToBeClickable(By.className("blank-slate-pf-main-action")));
+		projButton = driver.findElement(By.className("blank-slate-pf-main-action"));
```

Comment:
This seems to work only if this test is the first one executed because, otherwise, this "button" is not there in the web UI.
The problem is that there's no guarantee about the tests execution order so this test fails when executed not as the first one in a standard test execution.

Reply:
added annotations to fix test method execution order